nuxeo-features/nuxeo-elasticsearch/nuxeo-elasticsearch-core/src/main/java/org/nuxeo/elasticsearch/fetcher/VcsFetcher.java

```diff
@@ -126,6 +130,30 @@ public DocumentModelListImpl fetchDocuments() {
         return session.query(sb.toString());
     }
 
+    private void addHighlights(List<DocumentModel> docs) {
+        for (SearchHit hit : getResponse().getHits()) {
+            for (DocumentModel doc : docs) {
+                String docId = doc.getRepositoryName() + doc.getId();
+                String hitId = getRepoForIndex(hit.getIndex()) + hit.getId();
+                if (docId.equals(hitId)) {
+                    // Add highlight if it exists
+                    Map<String, HighlightField> esHighlights = hit.highlightFields();
+                    if (!esHighlights.isEmpty()) {
+                        Map<String, List<String>> fields = new HashMap<>();
+                        for (String field : esHighlights.keySet()) {
```

Comment:
Same comments as elsewhere about `entrySet()` and not re-fetching the `List` from `fields`.